scc/git.py

```diff
@@ -1007,7 +1007,8 @@ def find_candidate_branches(self, filters,
 
 class GitRepository(object):
 
-    def __init__(self, gh, path, remote="origin", repository_config=None):
+    def __init__(self, gh, path, remote="origin", push_branch="origin",
```

Comment:
In the case where `--push-branch` is not passed to the original command, am I correct that with the current implementation `scc merge <xxx> --update-gitmodules` would unconditionally write `branch = origin` in the `.gitmodules`?

My expectation would be that in that case, we might not want to modify this key i.e. set the default here to `None` and conditionally skip the setting of `submodule.<name>.branch`.

Reply:
Please see the additional commit.